pkg/pillar/cas/containerd.go

```diff
@@ -705,6 +708,82 @@ func (c *containerdCAS) PrepareContainerRootDir(rootPath, reference, rootBlobSha
 		logrus.Errorf(err.Error())
 		return err
 	}
+	// On kubevirt eve we also write the mountpoints and cmdline files to the rootPath.
+	// Once rootPath is populated with all necessary files, this rootPath directory will be
+	// converted to a PVC and passed in to domainmgr to attach to VM as rootdisk
+	// NOTE: For kvm eve these files are generated and passed to bootloader in pkg/pillar/containerd/oci.go:AddLoader()
+	if base.IsHVTypeKube() {
+
+		// Mount the snapshot
+		if err := c.MountSnapshot(snapshotID, GetRoofFsPath(rootPath)); err != nil {
+			return fmt.Errorf("PrepareContainerRootDir error mount of snapshot: %s. %s", snapshotID, err)
+		}
+
+		for m := range mountpoints {
+
+			b := []byte(fmt.Sprintf("%+v\n", m))
+			if err := os.WriteFile(filepath.Join(rootPath, "mountPoints"), b, 0666); err != nil {
+				err = fmt.Errorf("PrepareContainerRootDir: Exception while writing mountpoints to %v %v",
+					rootPath, err)
+				logrus.Errorf(err.Error())
+				return err
+			}
+		}
+
+		// create env manifest
+		envContent := ""
+		if workdir != "" {
+			envContent = fmt.Sprintf("export WORKDIR=\"%s\"\n", workdir)
+		} else {
+			envContent = fmt.Sprintf("export WORKDIR=/\n")
+		}
+		for _, e := range unProcessedEnv {
+			keyAndValueSlice := strings.SplitN(e, "=", 2)
+			if len(keyAndValueSlice) == 2 {
+				//handles Key=Value case
+				envContent = envContent + fmt.Sprintf("export %s=\"%s\"\n", keyAndValueSlice[0], keyAndValueSlice[1])
```

Comment:
@rene this reminds me about your PR (https://github.com/lf-edge/eve/pull/3760/files ) - can we use the same code base here?